src/typingsInstaller/nodeTypingsInstaller.ts

```diff
@@ -272,8 +227,50 @@ process.on("disconnect", () => {
     }
     process.exit(0);
 });
-const installer = new NodeTypingsInstaller(globalTypingsCacheLocation!, typingSafeListLocation!, typesMapLocation!, npmLocation, validateDefaultNpmLocation, /*throttleLimit*/5, log); // TODO: GH#18217
-installer.listen();
+let installer: NodeTypingsInstaller | undefined;
+process.on("message", (req: TypingInstallerRequestUnion) => {
+    installer ??= new NodeTypingsInstaller(globalTypingsCacheLocation!, typingSafeListLocation!, typesMapLocation!, npmLocation, validateDefaultNpmLocation, /*throttleLimit*/5, log); // TODO: GH#18217

```

Comment:
Rather than move this entire method body out + exposing a private property, can we instead move it to a method like `handleMessage`, and then at the bottom do something like:

```ts
let installer: NodeTypingsInstaller | undefined;
process.on("message", (req: TypingInstallerRequestUnion) => {
    installer ??= new NodeTypingsInstaller(globalTypingsCacheLocation!, typingSafeListLocation!, typesMapLocation!, npmLocation, validateDefaultNpmLocation, /*throttleLimit*/5, log); // TODO: GH#18217
    installer.handleMessage(req)
})
```